src/services/context-provide/ContextProvidePlugin.js

```diff
@@ -0,0 +1,40 @@
+import flow from 'lodash/flow'
+import camelCase from 'lodash/camelCase'
+import upperFirst from 'lodash/upperFirst'
+import cloneDeep from 'lodash/cloneDeep'
+import assignWith from 'lodash/assignWith'
+
+const pascalCase = flow(camelCase, upperFirst)
+export const ContextProviderPluginKey = 'va-context-config'
+
+class ContextProviderConfig {
+  constructor (options) {
+    this._config = options
+  }
+  getNewConfig (config) {
+    const _combinedConfig = assignWith(config, cloneDeep(this._config), (objVal) => objVal)// NOTE: need save the reference to config prop. Support reactivity
```

Comment:
// Merge external new config with current config.

Like this maybe?
Comment doesn't make a lot of sense to me.